sips/fork_support.md

```diff
@@ -20,56 +20,259 @@ Client implementation should calculate when a new duty starts to decide if it st
+
+
+```go
+// NetworkID are intended to separate different SSV networks. A network can have many forks in it.
+type NetworkID byte
 
 const (
-    GenesisChain = 0x0
-    PrimusChain  = 0x1
-    ShifuChain   = 0x2
-    JatoChain    = 0x3
+	MainnetNetworkID = NetworkID(0x0)
+	PrimusNetworkID  = NetworkID(0x1)
+	ShifuNetworkID   = NetworkID(0x2)
+	JatoNetworkID    = NetworkID(0x3)
 )
 
+// DomainType is a unique identifier for signatures, 2 identical pieces of data signed with different domains will result in different sigs
+type DomainType [4]byte
+
+// DomainTypes represent specific forks for specific chains, messages are signed with the domain type making 2 messages from different domains incompatible
 var (
-    GenesisMainnet = DomainType{0x0, 0x0, GenesisChain, 0x0}
-    PrimusTestnet  = DomainType{0x0, 0x0, PrimusChain, 0x0}
-    ShifuTestnet   = DomainType{0x0, 0x0, ShifuChain, 0x0}
-    ShifuV2Testnet = DomainType{0x0, 0x0, ShifuChain, 0x1}
-    JatoTestnet    = DomainType{0x0, 0x0, JatoChain, 0x1}
+	GenesisMainnet = DomainType{0x0, 0x0, byte(MainnetNetworkID), 0x0}
+	PrimusTestnet  = DomainType{0x0, 0x0, byte(PrimusNetworkID), 0x0}
+	ShifuTestnet   = DomainType{0x0, 0x0, byte(ShifuNetworkID), 0x0}
+	ShifuV2Testnet = DomainType{0x0, 0x0, byte(ShifuNetworkID), 0x1}
+	JatoTestnet    = DomainType{0x0, 0x0, byte(JatoNetworkID), 0x0}
+	JatoV2Testnet  = DomainType{0x0, 0x0, byte(JatoNetworkID), 0x1}
 )
 
+// ForkData is a simple structure holding fork information for a specific chain (and its fork)
 type ForkData struct {
-    Epoch  phase0.Epoch
-    Domain DomainType
+	// Epoch in which the fork happened
+	Epoch phase0.Epoch
+	// Domain for the new fork
+	Domain DomainType
+}
+
+func (domainType DomainType) GetNetworkID() NetworkID {
+	return NetworkID(domainType[2])
 }
 
-func (domainType DomainType) GetChain() byte {
-    return domainType[2]
+func (networkID NetworkID) GetForksData() []*ForkData {
+	switch networkID {
+	case MainnetNetworkID:
+		return mainnetForks()
+	default:
+		return []*ForkData{
+			{
+				Epoch:  0,
+				Domain: DomainType{0x0, 0x0, byte(networkID), 0x0},
+			},
+		}
+	}
 }
 
-func (domainType DomainType) GetForksData() []*ForkData {
-    switch domainType.GetChain() {
-    case GenesisChain:
-        return genesisForks()
-    default:
-        return []*ForkData{}
-    }
+// mainnetForks returns all forks for the mainnet chain
+func mainnetForks() []*ForkData {
+	return []*ForkData{
+		{
+			Epoch:  0,
+			Domain: GenesisMainnet,
+		},
+	}
 }
 
-func genesisForks() []*ForkData {
-    return []*ForkData{
-        {
-            Epoch:  0,
-            Domain: GenesisMainnet,
-        },
-    }
+func (networkID NetworkID) DefaultFork() *ForkData {
+	return networkID.GetForksData()[0]
 }
 
+// GetCurrentFork returns the ForkData with highest Epoch smaller or equal to "epoch"
+func (networkID NetworkID) GetCurrentFork(epoch phase0.Epoch) (*ForkData, error) {
```

Comment:
If we want to use GetCurrentFork name for this method, we should not pass epoch but get current epoch from beacon network
I would suggest to rename it to smtg like `ForkAtEpoch` or `ActiveForkAtEpoch`

Reply:
Sure, well observed. Thanks!